.github/actions/build/action.yml

```diff
@@ -39,17 +39,37 @@ runs:
         registry: ${{ inputs.docker_registry }}
         username: ${{ inputs.docker_username }}
         password: ${{ inputs.docker_password }}
+      
+    - name: Create ~/.npm Directory
+      shell: bash
+      run: mkdir -p ~/.npm
+
+    - name: Cache node modules
+      id: cache-npm
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ inputs.context }}
+        restore-keys: |
+          ${{ inputs.context }}
 
     - name: Docker build
       id: build
-      uses: docker/build-push-action@v2
+      uses: docker/build-push-action@v4
       with:
         context: ${{ inputs.context }}
         tags: ${{ inputs.tags }}
         build-args: ${{ inputs.build_args }}
-        # @see - https://docs.docker.com/build/ci/github-actions/cache/#github-cache
-        cache-from: type=gha
-        cache-to: type=gha,mode=max
+        cache-from: type=gha, scope=$GITHUB_REF_NAME-image2
+        cache-to: type=gha, key={{ $inputs.context }} mode=max, scope=$GITHUB_REF_NAME-image2
         outputs: type=image,push=true
-        # @see - https://docs.docker.com/build/ci/github-actions/secrets/
+        push: true
```

Comment:
If `push` is mentioned in `outputs`, don't think you need to additionally provide it. Kindly remove this.